File: app/src/main/java/com/example/pc_0775/naugthyvideo/base/ActivityCollector.java
package com.example.pc_0775.naugthyvideo.base;

import android.app.Activity;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by PC-0775 on 2018/5/29.
 */

public class ActivityCollector {

    /**
     * Activity集合
     **/
    public static List<Activity> activities = new ArrayList<>();

    public static void addActivity(Activity activity) {
        activities.add(activity);
    }

    public static void removeActivity(Activity activity) {
        activities.remove(activity);
    }

    public static void finishAll() {
        for (Activity activity : activities) {
            if (!activity.isFinishing()) {
                activity.finish();
            }
        }
    }
}

File: app/src/main/java/com/example/pc_0775/naugthyvideo/app/ActivityCollector.kt
package com.example.pc_0775.naugthyvideo.app

import android.app.Activity

/**
 * Created by PC-0775 on 2018/12/25.
 */
class ActivityCollector {
    companion object {
        var activities = ArrayList<Activity>()

        fun addActivity(activity: Activity){
            activities.add(activity)
        }

        fun removeActivity(activity: Activity){
            activities.remove(activity)
        }

        fun finishAll(){
            for(activity in activities){
                if (!activity.isFinishing){
                    activity.finish()
                }
            }
        }
    }
}